### ISSUE TITLE
PS5 question

### ISSUE THREAD

**Issue by Helba0** [Soreepeong/XivMitmLatencyMitigator#88]
Hi, i’m interested on using this for a ps5. This is all very new to me and I’m at loss where to start, and I only have a Mac on me. Is it still possible? 
I have some questions: 

1. Where do I start? Do I need to have some sort of a bootcamp partition so that I could run windows program on mac?
2. Do I need to download the game for mac first? Will the free trial version for mac be sufficient? 
3. Will this affect my online experience for other games on the PS5? How easy is it to ‘technically’ turn it on and off?

I’m pretty sure these are all amateur questions, but I hope I can have some answers to them. I have no trouble _trying_ to follow the instructions, though I’m sure I’ll make mistakes here and there. I just hope to have a starting point because I don’t know where to start. Thank you very much and I’m sorry for the inconvenience.

**Comment by bankjaneo** [comment]
1. You don't need Bootcamp to run any Windows or macOS program.

2. You also don't need to download any version of the game. All you need for this script to work is Linux virtual machine and ffxiv_dx11.exe file from PC version.

3. When everything is done, it will affect PS5 Internet connection since it will be route through your Mac but you can always revert it anytime on your PS5 network settings.

To get this thing working, first, I'm not sure if this script can run natively on Mac or not, especially on Mac with Apple Silicon.

Basically, you need to create a Linux virtual machine (VM) like Ubuntu 22.04 or Debian 12. You can use free software like VirtualBox to do this and don't forget to set network adapter to "Bridge" so your VM can get its own IP. Once you get this, the rest of instructions is on the first page of this repo.

Btw, I believe Mac with Apple Silicon is not able to get this script running because VM will be a ARM version of Linux.

**Comment by bankjaneo** [comment]
Yes, once you have a VM, you can skip to step 5.

For the ffxiv_dx11.exe, you can always get it from PC version of the game. But if you don't have it and you trust random guy on the Internet, you can get 6.40 version of that file at https://github.com/bankjaneo/XivMitmDocker/tree/main/app. I haven't update to 6.45 yet but even 6.40 can also get the script running without any problem.

**Comment by Helba0** [comment]
I'd rather trust a random guy on the intenet :)
does the ffxiv_dx11.exe serves as a replacement for ffxiv.exe in step 7?

**Comment by bankjaneo** [comment]
That's correct!

**Comment by Helba0** [comment]
I ended up using Debian 11, and on step 6 this is what i get.

![error 1](https://github.com/Soreepeong/XivMitmLatencyMitigator/assets/139856795/5aad31a6-6820-4d5c-94a7-b258267afa4a)

Am i doing something wrong? maybe it's the network settings. this is how i set it up.
![network settings](https://github.com/Soreepeong/XivMitmLatencyMitigator/assets/139856795/23230f34-ef02-40ee-9f51-1b8670abf4ee)

**Comment by bankjaneo** [comment]
You need to remove Angle Brackets < and > from both IP address and interface name.

**Comment by Helba0** [comment]
Ah i'm sorry. I tried it but this time it says iptables command not found. I tried gooling to install the command, but nothing is currently working as of now..

**Comment by Helba0** [comment]
An update, I'd like to think that step 6 worked, so I moved on to step 7. I installed wine on the debian so i could open the HFS file. This is what i got so far.
![Step 7](https://github.com/Soreepeong/XivMitmLatencyMitigator/assets/139856795/c6d3c5d2-8d22-4028-be81-b94c84f049e3)

And then I proceeded to step 8, which looked like it worked because i got no warning about the storage thing.
I went to step 9 but this is what i got.

![Step 8](https://github.com/Soreepeong/XivMitmLatencyMitigator/assets/139856795/8ff2b316-8e9f-44f8-9d9e-4648afa613d4)

it says python command not found. still looking for solutions to install python.

**Comment by bankjaneo** [comment]
Successul `iptables` command will give no result so you have completed that step.

For curl command, it needs to be `-o` instead of `-0` (o stands for output), then it will save the downloaded file as `ffxiv_dx11.exe` as commanded.

The python one you can change `| python` at the end of command to `| python3` and see if it works.

You can get `<vm_ip>` from `ip addr show` command. From the above screenshot, your `<vm_ip>` is `192.168.68.115`

**Comment by Helba0** [comment]
Okay. i redo step 7 and here's what i got.

![step7b](https://github.com/Soreepeong/XivMitmLatencyMitigator/assets/139856795/56895ba1-7d3d-4eac-bf90-a2d85c9bd044)

and this is for step 9.

![step 9b](https://github.com/Soreepeong/XivMitmLatencyMitigator/assets/139856795/54a01eb2-7c56-4bfd-b1b8-e892bc633259)

I'll try doing the rest of the steps.

just a question in advance, what does the network routing looks like? currently I have a router with an ethernet connection to the macbook, and a ps5 with a wireless connection.

**Comment by bankjaneo** [comment]
Everything almost looking fine now except the script exited after run.

Did you install libraries from the step 8 yet?

`apt install g++ gcc-multilib g++-multilib`

**Comment by bankjaneo** [comment]
Network routing will look like this after you set PS5 network gateway.

Internet <--> Router <--> VM <--> PS5

**Comment by Helba0** [comment]
ahh i skipped step 8 because i thought i did it right before. it still posted the same thing i think. 

![error 3](https://github.com/Soreepeong/XivMitmLatencyMitigator/assets/139856795/075d561c-354b-4f20-a265-abbd2dd4a6e7)

does the PS5 need a wired connection from the macbook? or can I use a wired connection from the router as well?

**Comment by bankjaneo** [comment]
Can you show me the output of `ls -l` ?

**Comment by Helba0** [comment]
root@debian:/home/orca/Python-3.10.0# ls -l
total 76752
-rw-r--r--  1 orca orca    10121 Oct  4  2021 aclocal.m4
drwxr-xr-x  5 root root     4096 Jul 18 20:15 build
-rw-r--r--  1 orca orca      630 Oct  4  2021 CODE_OF_CONDUCT.md
-rwxr-xr-x  1 orca orca    44166 Oct  4  2021 config.guess
-rw-r--r--  1 root root   651960 Jul 18 20:08 config.log
-rwxr-xr-x  1 root root    43040 Jul 18 20:08 config.status
-rwxr-xr-x  1 orca orca    36251 Oct  4  2021 config.sub
-rwxr-xr-x  1 orca orca   517424 Oct  4  2021 configure
-rw-r--r--  1 orca orca   177074 Oct  4  2021 configure.ac
drwxr-xr-x 18 orca orca     4096 Oct  4  2021 Doc
-rw-r--r--  1 root root     2361 Jul 18 20:46 ffxiv_dx11.exe
drwxr-xr-x  2 orca orca     4096 Oct  4  2021 Grammar
drwxr-xr-x  4 orca orca     4096 Oct  4  2021 Include
-rwxr-xr-x  1 orca orca    15368 Oct  4  2021 install-sh
drwxr-xr-x 35 orca orca     4096 Jul 18 20:20 Lib
-rw-r--r--  1 root root 52160394 Jul 18 20:20 libpython3.10.a
-rw-r--r--  1 orca orca    13925 Oct  4  2021 LICENSE
drwxr-xr-x  8 orca orca     4096 Oct  4  2021 Mac
-rw-r--r--  1 root root    81941 Jul 18 20:08 Makefile
-rw-r--r--  1 root root    72831 Jul 18 20:08 Makefile.pre
-rw-r--r--  1 orca orca    73055 Oct  4  2021 Makefile.pre.in
drwxr-xr-x  2 orca orca     4096 Jul 18 20:08 Misc
drwxr-xr-x 14 orca orca     4096 Jul 18 20:20 Modules
-rw-r--r--  1 orca orca       82 Oct  4  2021 netlify.toml
drwxr-xr-x  4 orca orca     4096 Jul 18 20:18 Objects
drwxr-xr-x  2 orca orca     4096 Jul 18 20:16 Parser
drwxr-xr-x  5 orca orca     4096 Oct  4  2021 PC
drwxr-xr-x  2 orca orca     4096 Oct  4  2021 PCbuild
-rw-r--r--  1 root root        0 Jul 18 20:16 profile-run-stamp
drwxr-xr-x  2 orca orca     4096 Jul 18 20:20 Programs
-rw-r--r--  1 root root       27 Jul 18 20:20 pybuilddir.txt
-rw-r--r--  1 root root    48398 Jul 18 19:50 pyconfig.h
-rw-r--r--  1 orca orca    46258 Oct  4  2021 pyconfig.h.in
-rwxr-xr-x  1 root root 24281168 Jul 18 20:20 python
drwxr-xr-x  3 orca orca     4096 Jul 18 20:20 Python
-rw-r--r--  1 root root     3090 Jul 18 20:09 python-config
-rw-r--r--  1 root root     2044 Jul 18 20:09 python-config.py
-rw-r--r--  1 root root    65722 Jul 18 20:09 python-gdb.py
-rw-r--r--  1 orca orca    10138 Oct  4  2021 README.rst
-rw-r--r--  1 orca orca   115949 Oct  4  2021 setup.py
drwxr-xr-x 23 orca orca     4096 Oct  4  2021 Tools
root@debian:/home/orca/Python-3.10.0# 

I tried the curl command again without the sudo, it posted the same thing. above is the output of ls -l

**Comment by bankjaneo** [comment]
I think the problem is from incomplete download of `ffxiv_dx11.exe` file. The file size is only 2361 Bytes instead of 43265112 Bytes

`-rw-r--r-- 1 root root 2361 Jul 18 20:46 ffxiv_dx11.exe`

Try delete the old file and use this command to download file directly from my github repo.

`rm ffxiv_dx11.exe`
`curl -o ffxiv_dx11.exe https://raw.githubusercontent.com/bankjaneo/XivMitmDocker/main/app/ffxiv_dx11.exe`

**Comment by Helba0** [comment]
ah okay. I'll try doing that. I assume I need to redo step 7 (HFS) through step 9?

**Comment by bankjaneo** [comment]
No, if you run this command directly in your VM, you can skip step 7

**Comment by Helba0** [comment]
Okay. I downloaded the file directly, it posted a different thing now.

root@debian:/home/orca/Python-3.10.0# rm ffxiv_dx11.exe
root@debian:/home/orca/Python-3.10.0# curl -o ffxiv_dx11.exe https://raw.githubusercontent.com/bankjaneo/XivMitmDocker/main/app/ffxiv_dx11.exe
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 41.2M  100 41.2M    0     0   265k      0  0:02:39  0:02:39 --:--:--  239k
root@debian:/home/orca/Python-3.10.0# install g++ gcc-multilib g++-multilib
install: target 'g++-multilib' is not a directory
root@debian:/home/orca/Python-3.10.0# apt install g++ gcc-multilib g++-multilib
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
g++ is already the newest version (4:10.2.1-1).
g++-multilib is already the newest version (4:10.2.1-1).
gcc-multilib is already the newest version (4:10.2.1-1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
root@debian:/home/orca/Python-3.10.0# curl https://raw.githubusercontent.com/Soreepeong/XivMitmLatencyMitigator/main/mitigate.py | python3
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  109k  100  109k    0     0   116k      0 --:--:-- --:--:-- --:--:--  115k
2023-07-19 00:30:29,677	17278(main)	INFO	Region filter: (None)
2023-07-19 00:30:29,678	17278(main)	INFO	Extra delay: 0.075s
2023-07-19 00:30:29,678	17278(main)	INFO	Use measured socket latency: no
2023-07-19 00:30:32,403	17278(main)	INFO	Downloading opcode definition files...
2023-07-19 00:30:33,677	17278(main)	INFO	Running: iptables -t nat -I PREROUTING -p tcp -m multiport --dports 1025:65535 -d 202.67.52.0/24 -j REDIRECT --to 57307
sh: 1: iptables: not found
2023-07-19 00:30:33,682	17278(main)	ERROR	This program requires root permissions.

2023-07-19 00:30:33,682	17278(main)	INFO	Cleaning up...
2023-07-19 00:30:33,682	17278(main)	ERROR	One or more error have occurred during cleanup.
root@debian:/home/orca/Python-3.10.0# 

I think there's still an error. did the iptables not work?

**Comment by bankjaneo** [comment]
That's a little strange since you are root and it still asks for root permissions. Try this command instead.

`curl https://raw.githubusercontent.com/Soreepeong/XivMitmLatencyMitigator/main/mitigate.py | sudo python3`

**Comment by Helba0** [comment]
root@debian:/home/orca/Python-3.10.0# curl https://raw.githubusercontent.com/Soreepeong/XivMitmLatencyMitigator/main/mitigate.py | sudo python3
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  109k  100  109k    0     0   218k      0 --:--:-- --:--:-- --:--:--  218k
2023-07-19 00:36:38,326	17315(main)	INFO	Region filter: (None)
2023-07-19 00:36:38,326	17315(main)	INFO	Extra delay: 0.075s
2023-07-19 00:36:38,327	17315(main)	INFO	Use measured socket latency: no
2023-07-19 00:36:41,292	17315(main)	INFO	Running: iptables -t nat -I PREROUTING -p tcp -m multiport --dports 1:65535 -d 119.97.0.0/16 -j REDIRECT --to 23469
2023-07-19 00:36:41,539	17315(main)	INFO	Running: iptables -t nat -I PREROUTING -p tcp -m multiport --dports 1025:65535 -d 153.254.80.0/24 -j REDIRECT --to 23469
2023-07-19 00:36:41,546	17315(main)	INFO	Running: iptables -t nat -I PREROUTING -p tcp -m multiport --dports 1025:65535 -d 183.111.189.0/24 -j REDIRECT --to 23469
2023-07-19 00:36:41,549	17315(main)	INFO	Running: iptables -t nat -I PREROUTING -p tcp -m multiport --dports 1025:65535 -d 124.150.157.0/24 -j REDIRECT --to 23469
2023-07-19 00:36:41,552	17315(main)	INFO	Running: iptables -t nat -I PREROUTING -p tcp -m multiport --dports 1:65535 -d 162.14.0.0/16 -j REDIRECT --to 23469
2023-07-19 00:36:41,570	17315(main)	INFO	Running: iptables -t nat -I PREROUTING -p tcp -m multiport --dports 1025:65535 -d 80.239.145.0/24 -j REDIRECT --to 23469
2023-07-19 00:36:41,586	17315(main)	INFO	Running: iptables -t nat -I PREROUTING -p tcp -m multiport --dports 1025:65535 -d 204.2.29.0/24 -j REDIRECT --to 23469
2023-07-19 00:36:41,597	17315(main)	INFO	Running: iptables -t nat -I PREROUTING -p tcp -m multiport --dports 1025:65535 -d 202.67.52.0/24 -j REDIRECT --to 23469
2023-07-19 00:36:41,620	17315(main)	INFO	Running: iptables -t nat -I PREROUTING -p tcp -m multiport --dports 1:65535 -d 27.221.0.0/16 -j REDIRECT --to 23469
2023-07-19 00:36:41,625	17315(main)	INFO	Running: iptables -t nat -I PREROUTING -p tcp -m multiport --dports 1:65535 -d 109.244.0.0/16 -j REDIRECT --to 23469
2023-07-19 00:36:41,648	17315(main)	INFO	Running: iptables -t nat -I PREROUTING -p tcp -m multiport --dports 1025:65535 -d 210.94.0.0/24 -j REDIRECT --to 23469
net.ipv4.ip_forward = 1
net.ipv4.conf.all.route_localnet = 1
2023-07-19 00:36:41,739	17315(main)	INFO	Listening on ('0.0.0.0', 23469)...
2023-07-19 00:36:41,739	17315(main)	INFO	Press Ctrl+C to quit.

That seemed to do the trick? for step 10 i assume i should run it out of the VM and with the macbook terminal yes? for the (vm_ip) do i just use the 192.168.68.117 from below without the /24?

orca@debian:~$ ip addr show
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: enp0s3: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether 08:00:27:a3:c7:04 brd ff:ff:ff:ff:ff:ff
    inet 192.168.68.117/24 brd 192.168.68.255 scope global dynamic noprefixroute enp0s3
       valid_lft 6938sec preferred_lft 6938sec
    inet6 fe80::a00:27ff:fea3:c704/64 scope link noprefixroute 
       valid_lft forever preferred_lft forever

**Comment by bankjaneo** [comment]
Yes, yes and yes 😁

**Comment by Helba0** [comment]
okay, so from what i understand, I need to connect a wired cable from the macbook to the ps5?  any chance that this could be done wireless?

**Comment by bankjaneo** [comment]
You don't need that. Leave everything as it is and just set the gateway in your PS5 network settings to VM IP.

**Comment by Helba0** [comment]
the default gateway must be set to <vm_ip>, does these apply to the IPv4 Address and the Subnet Mask too?

**Comment by bankjaneo** [comment]
I'm not familiar with PS5 settings but technically, you need to change ONLY gateway setting. Leave IPv4 and Subnet mask as it is.

**Comment by Helba0** [comment]
Omg it felt like it.. works?? everything feels smoother. you're a legend man. Thank you so much for being so patient with me. 
How do I turn it off / on? Do I just need to turn on the macbook? Do I have to open the debian every session?

This is the feedback from the terminal just for reference.
2023-07-19 01:26:25,325	17747	INFO	C2S_ActionRequest: actionId=5f3f sequence=0063
2023-07-19 01:26:25,445	17747	INFO	S2C_ActionEffect: actionId=5f3f sourceSequence=0063 wait=600ms->554ms rtt=120ms
2023-07-19 01:26:26,203	17747	INFO	S2C_ActionEffect: actionId=0007 sourceSequence=0000 wait=100ms serverOriginated
2023-07-19 01:26:27,829	17747	INFO	C2S_ActionRequest: actionId=5f3e sequence=0064
2023-07-19 01:26:27,938	17747	INFO	S2C_ActionEffect: actionId=5f3e sourceSequence=0064 wait=600ms->566ms rtt=108ms
2023-07-19 01:26:29,464	17747	INFO	S2C_ActionEffect: actionId=0007 sourceSequence=0000 wait=100ms serverOriginated
2023-07-19 01:26:30,347	17747	INFO	C2S_ActionRequest: actionId=5f35 sequence=0065
2023-07-19 01:26:30,541	17747	INFO	S2C_ActionEffect: actionId=5f35 sourceSequence=0065 wait=600ms->480ms rtt=195ms
2023-07-19 01:26:32,613	17747	INFO	S2C_ActionEffect: actionId=0007 sourceSequence=0000 wait=100ms serverOriginated
2023-07-19 01:26:48,067	17747	INFO	C2S_ActionRequest: actionId=1cf3 sequence=0067
2023-07-19 01:26:48,174	17747	INFO	S2C_ActionEffect: actionId=1cf3 sourceSequence=0067 wait=600ms->567ms rtt=107ms
2023-07-19 01:26:48,175	17747	INFO	S2C_ActionEffect: actionId=0008 sourceSequence=0000 wait=600ms->567ms serverOriginated
2023-07-19 01:26:50,518	17747	INFO	C2S_ActionRequest: actionId=1cf4 sequence=0068
2023-07-19 01:26:50,634	17747	INFO	S2C_ActionEffect: actionId=1cf4 sourceSequence=0068 wait=600ms->558ms rtt=116ms
2023-07-19 01:26:50,855	17747	INFO	S2C_ActionEffect: actionId=0008 sourceSequence=0000 wait=376ms->335ms serverOriginated
2023-07-19 01:26:52,990	17747	INFO	C2S_ActionRequest: actionId=1cf5 sequence=0069
2023-07-19 01:26:53,119	17747	INFO	S2C_ActionEffect: actionId=1cf5 sourceSequence=0069 wait=600ms->545ms rtt=130ms
2023-07-19 01:26:53,521	17747	INFO	S2C_ActionEffect: actionId=0008 sourceSequence=0000 wait=199ms->145ms serverOriginated
2023-07-19 01:26:54,407	17747	INFO	C2S_ActionRequest: actionId=1cf6 sequence=006a
2023-07-19 01:26:54,546	17747	INFO	S2C_ActionEffect: actionId=1cf6 sourceSequence=006a wait=600ms->536ms rtt=139ms
2023-07-19 01:26:55,275	17747	INFO	C2S_ActionRequest: actionId=0b3e sequence=006b
2023-07-19 01:26:55,406	17747	INFO	S2C_ActionEffect: actionId=0b3e sourceSequence=006b wait=600ms->544ms rtt=131ms
2023-07-19 01:26:55,957	17747	INFO	C2S_ActionRequest: actionId=4072 sequence=006c
2023-07-19 01:26:56,161	17747	INFO	S2C_ActionEffect: actionId=4072 sourceSequence=006c wait=600ms->471ms rtt=204ms
2023-07-19 01:26:56,259	17747	INFO	S2C_ActionEffect: actionId=0008 sourceSequence=0000 wait=509ms->381ms serverOriginated
2023-07-19 01:26:56,691	17747	INFO	C2S_ActionRequest: actionId=4339 sequence=006d
2023-07-19 01:26:56,816	17747	INFO	S2C_ActionEffect: actionId=4339 sourceSequence=006d wait=600ms->550ms rtt=124ms
2023-07-19 01:26:58,447	17747	INFO	C2S_ActionRequest: actionId=1cf2 sequence=006e
2023-07-19 01:26:58,594	17747	INFO	S2C_ActionEffect: actionId=1cf2 sourceSequence=006e wait=600ms->528ms rtt=147ms
2023-07-19 01:26:58,860	17747	INFO	S2C_ActionEffect: actionId=0008 sourceSequence=0000 wait=332ms->261ms serverOriginated
2023-07-19 01:26:59,947	17747	INFO	C2S_ActionRequest: actionId=1cf2 sequence=006f
2023-07-19 01:27:00,064	17747	INFO	S2C_ActionEffect: actionId=1cf2 sourceSequence=006f wait=600ms->557ms rtt=117ms
2023-07-19 01:27:00,629	17747	INFO	C2S_ActionRequest: actionId=0b4a sequence=0070
2023-07-19 01:27:00,824	17747	INFO	S2C_ActionEffect: actionId=0b4a sourceSequence=0070 wait=600ms->479ms rtt=195ms
2023-07-19 01:27:01,447	17747	INFO	C2S_ActionRequest: actionId=1cf2 sequence=0071
2023-07-19 01:27:01,531	17747	INFO	S2C_ActionEffect: actionId=0008 sourceSequence=0000 wait=100ms serverOriginated
2023-07-19 01:27:01,627	17747	INFO	S2C_ActionEffect: actionId=1cf2 sourceSequence=0071 wait=600ms->495ms rtt=180ms
2023-07-19 01:27:02,117	17747	INFO	C2S_ActionRequest: actionId=0b3a sequence=0072
2023-07-19 01:27:02,293	17747	INFO	S2C_ActionEffect: actionId=0b3a sourceSequence=0072 wait=600ms->504ms rtt=176ms
2023-07-19 01:27:02,950	17747	INFO	C2S_ActionRequest: actionId=1cf2 sequence=0073
2023-07-19 01:27:03,143	17747	INFO	S2C_ActionEffect: actionId=1cf2 sourceSequence=0073 wait=600ms->482ms rtt=193ms
2023-07-19 01:27:03,615	17747	INFO	C2S_ActionRequest: actionId=0b4a sequence=0074
2023-07-19 01:27:03,754	17747	INFO	S2C_ActionEffect: actionId=0b4a sourceSequence=0074 wait=600ms->546ms rtt=139ms
2023-07-19 01:27:04,195	17747	INFO	S2C_ActionEffect: actionId=0008 sourceSequence=0000 wait=156ms->102ms serverOriginated
2023-07-19 01:27:04,451	17747	INFO	C2S_ActionRequest: actionId=1cf2 sequence=0075
2023-07-19 01:27:04,594	17747	INFO	S2C_ActionEffect: actionId=1cf2 sourceSequence=0075 wait=600ms->532ms rtt=143ms
2023-07-19 01:27:05,118	17747	INFO	C2S_ActionRequest: actionId=0b3a sequence=0076
2023-07-19 01:27:05,314	17747	INFO	S2C_ActionEffect: actionId=0b3a sourceSequence=0076 wait=600ms->487ms rtt=197ms

**Comment by bankjaneo** [comment]
Congratulation! It is working flawlessly.

To stop, you just press `Ctrl+C` to stop the script and then you can safely shutdown VM. But when the VM is not running, you need to set PS5 network gateway back to your router IP (192.168.68.1 I think?) or your PS5 will not be able to connect to the Internet until you turn your VM on again.

Next time you want to use it again, turn on VM and go into the path that you have ffxiv_dx11.exe downloaded and run the script again. To summarize all your setup in 1 command should be

`cd /home/orca/Python-3.10.0/ && curl https://raw.githubusercontent.com/Soreepeong/XivMitmLatencyMitigator/main/mitigate.py | sudo python3`

And also don't forget to set PS5 network gateway to your VM IP.

**Comment by Helba0** [comment]
I see. I'll try powering everything off and try to turn it on again. Do I need to keep the connection wired to my macbook or it doesn't matter at all? I'll let you know if I managed to run everything back on.

**Comment by bankjaneo** [comment]
You can use wireless on your MacBook instead but it'll cost you with increase a little latency when gaming. The traffic flow when using this script will be like this.

Internet <--> Router <--> VM <--> Router <--> PS5

Every has its own increase in latency, but you can try to see if the increased latency is noticeable or not. 

One more thing, if you change from wire to wireless on your MacBook, don't forget to change Bridged Adapter `Name` in VirtualBox VM setting from Ethernet to Wi-Fi.

![image](https://github.com/Soreepeong/XivMitmLatencyMitigator/assets/14077691/4e360c86-a381-4c36-9747-e0eaacba5275)

**Comment by Helba0** [comment]
I just tried turning it off and on. Your command above works! also thank you for reminding me about the bridge adapter. Thank you so much! You've been very helpful. I have no further questions. Hopefully this could also help other clueless users like me who wanted to try out the mod on console :)

**Comment by bankjaneo** [comment]
Always happy to help fellow raiders. Don't forget mark the issue as Closed. Happy raiding!
